### DISCUSSION TITLE
error: Uncaught (in promise) Error: @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.

### DISCUSSION THREAD

**Discussion by mabdinasira** [prisma/prisma#25561] [Q&A] [answered]
### Question

Anyone know how to resolve the above error? I have gone through lots of steps, liek deleting node modules, deleting lock file,running npx prisma generate and more. 

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
generator client {
  provider        = "prisma-client-js"
  output          = "../generated/client"
  previewFeatures = ["deno"]
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model User {
  id         String   @id @default(cuid())
  firstName   String
  lastName   String
  mobile     String   @unique
  password   String
  createdAt  DateTime @default(now())
  updatedAt  DateTime @updatedAt
}

```

```ts
// Add any relevant Prisma Client queries here
const existingUser = await prisma.user.findUnique({
	where: { mobile },
});
```

- OS:  Mac OSX
- Database:  Supabase Pg
- Node.js version:  v23.1.0

- Run `prisma -v` to see your Prisma version and paste it: 5.21.1

**Comment by nurul3101** [comment]
Hello @mabdinasir 👋 

Did you get a confirmation message like this when you ran `npx prisma generate` command?

```txt
✔ Generated Prisma Client (v5.21.1) to ./node_modules/@prisma/client in 57ms

Start by importing your Prisma Client (See: http://pris.ly/d/importing-client)
```

**Reply by mabdinasira** [reply]
yes mate, I did get that.

<img width="957" alt="Screenshot 2024-11-05 at 12 00 37 am" src="https://github.com/user-attachments/assets/eca49a79-71bb-4236-aa70-e20d41294019">

**Reply by nurul3101** [reply]
Can you remove these two lines from your `schema.prisma` file?

```
  output          = "../generated/client"
  previewFeatures = ["deno"]
```

Run `npx prisma generate` command again and then try running the PrismaClient query.

I assume you are using Node.js and not Deno.

**Reply by mabdinasira** [reply]
no, I am actually using Deno. I will probably ditch it for node, I thought Deno 2.0 would be stable enough but I guess it isn't.

**Reply by nurul3101** [reply]
Did using Node.js fixed the issue for you?

**Reply by mabdinasira** [reply, marked as answer]
Yes, I switched to Node, no issues at all.

**Reply by nurul3101** [reply]
Glad to hear that! 👍
